### PR TITLE
Fix codebase and enable readability-* as clang-tidy check

### PR DESCRIPTION
**Pull request by david-alvarez-rosa**

### CODE REVIEW

File: include/qfm/pricing/pricing.hpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace pricing {
 class Pricing {
  public:
   void SetModel(std::shared_ptr<model::Model> model,
-                const asset::AssetType type) noexcept;
+                asset::AssetType type) noexcept;
```
Comment (waifod) [inline include/qfm/pricing/pricing.hpp]: Why did you remove `const` here?

File: src/qfm/pricing/pricing.cpp [hunk 2]
```diff
@@ -23,9 +23,9 @@ void Pricing::SetModel(std::shared_ptr<model::Model> model,
 
 double Pricing::GetAssetPrice(
     std::shared_ptr<asset::Asset> asset) const noexcept {
-  auto it = models_.find(asset->GetType());
-  if (it != models_.end()) {
-    return it->second->GetAssetPrice(asset);
+  if (auto iterator = models_.find(asset->GetType());
+      iterator != models_.end()) {
```
Comment (waifod) [inline src/qfm/pricing/pricing.cpp]: I'm not sure I like it because it looks less readable. Can you point to the best practices?

File: include/qfm/pricing/pricing.hpp (hunk 1 above)
Comment (david-alvarez-rosa) [reply include/qfm/pricing/pricing.hpp]: That's a good question, I asked myself the same when I was looking at the changes. The warning given by `clang-tidy` is the following

```txt
/Volumes/workplace/quant_finance_models/include/qfm/pricing/pricing.hpp:22:17: error: parameter 'type' is const-qualified in the function declaration; const-qualification of parameters only has an effect in function definitions [readability-avoid-const-params-in-decls,-warnings-as-errors]
   22 |                 const asset::AssetType type) noexcept;
      |                 ^~~~~
```

The reason is (I believe) that `type` is passed by value (it's copied), hence marking it as `const` in the **header** (`.hpp`) file adds not value to the consumer (is a implementation detail). In this case seems that best practice is to remove unnecessary information to the consumer, and maintain those implementation details only the `.cpp` file (you can check that the `const` is still there). This is the case when passing parameters by value, not by reference.

btw, this removal (and quite a few others) was automatically done by `clang-tidy` with `--fix` tag.

```prompt
$ find src include main.cpp -name '*.cpp' -o -name '*.hpp' | xargs -n1 clang-tidy --fix 
```

File: src/qfm/pricing/pricing.cpp (hunk 2 above)
Comment (david-alvarez-rosa) [reply src/qfm/pricing/pricing.cpp]: Reverted to minimize changes — only thing `clang-tidy` was marking as warning was short variable name, hence I renamed from `it` to `iterator`.

```text
/Volumes/workplace/quant_finance_models/src/qfm/pricing/pricing.cpp:26:8: error: variable name 'it' is too short, expected at least 3 characters [readability-identifier-length,-warnings-as-errors]
   26 |   auto it = models_.find(asset->GetType());
      |        ^
```

File: include/qfm/pricing/pricing.hpp (hunk 1 above)
Comment (waifod) [reply include/qfm/pricing/pricing.hpp]: What about making it a constant reference instead? No reason to copy, I think.

File: src/qfm/pricing/pricing.cpp (hunk 2 above)
Comment (waifod) [reply src/qfm/pricing/pricing.cpp]: It makes sense, thanks.

File: include/qfm/pricing/pricing.hpp (hunk 1 above)
Comment (david-alvarez-rosa) [reply include/qfm/pricing/pricing.hpp]: Yeah, that makes sense to me. Done!

**Review by waifod — APPROVED** [review]
LGTM! Approving and merging.
